app/src/main/java/com/github/pockethub/android/ui/notification/NotificationListAdapter.java

```diff
@@ -0,0 +1,127 @@
+                addItem(ITEM_HEADER, thread.repository());
+            }
+
+            addItem(ITEM_NOTIFICATION, thread);
+            repoFound = thread.repository();
+        }
+        notifyDataSetChanged();
+    }
+
+    @Override
+    public int getViewTypeCount() {
+        return 2;
+    }
+
+    @Override
+    protected int getChildLayoutId(int viewType) {
+        switch (viewType) {
+            case ITEM_NOTIFICATION:
+                return R.layout.notification_item;
+            case ITEM_HEADER:
+                return R.layout.notification_item_header;
+            default:
+                throw new IllegalStateException("Incorrect view type");
+        }
+    }
+
+    @Override
+    protected int[] getChildViewIds(int viewType) {
+        switch (viewType) {
+            case ITEM_NOTIFICATION:
+                return new int[] {R.id.tv_notification_icon, R.id.tv_notification_title,
+                                  R.id.tv_notification_date, R.id.tv_notification_read_icon};
+            case ITEM_HEADER:
+                return new int[] {R.id.tv_name, R.id.tv_notifications_read_icon};
+            default:
+                throw new IllegalStateException("Incorrect view type");
+        }
+    }
+
+    @Override
+    protected void update(int position, Object object, int viewType) {
+
+        if (viewType == ITEM_HEADER) {
+            // The header should not be clickable
+            updater.view.setEnabled(false);
+            updater.view.setOnClickListener(null);
+
+            Repository repository = (Repository) object;
+            setText(0, repository.fullName());
+
+            setText(1, "\uf03a");
+            if (!textView(1).hasOnClickListeners()) {
+                textView(1).setOnClickListener(v ->
+                        notificationReadListener.readNotifications(repository));
+            }
+        } else if (viewType == ITEM_NOTIFICATION) {
+            NotificationThread notificationThread = (NotificationThread) object;
+
+            String type = notificationThread.subject().type();
+            switch (type) {
+                case "Issue":
+                    setText(0, OcticonTextView.ICON_ISSUE_OPEN);
+                    break;
+                case "Release":
+                    setText(0, OcticonTextView.ICON_TAG);
+                    break;
+                default:
+                    setText(0, OcticonTextView.ICON_PULL_REQUEST);
+                    break;
+            }
+
+            setText(1, notificationThread.subject().title());
+            setText(2, TimeUtils.getRelativeTime(notificationThread.updatedAt()));
+
+            setText(3, "\uf03a");
+            setGone(3, !notificationThread.unread());
+
+            if (!textView(3).hasOnClickListeners()) {
+                textView(3).setOnClickListener(v ->
+                        notificationReadListener.readNotification(notificationThread));
```

Comment:
Can we also call the read notification API when clicking on list item? This may be a better user experience on mobile, every time user click the notification and read detail page, app will mark read automatically for them.